### PR TITLE
Refactor fastExponent.go

### PR DESCRIPTION
**Pull request by vedantmamgain**
refactor: condensed the function and added comments for the steps

### CODE REVIEW

File: math/fastExponent.go [hunk 1]
```diff
@@ -1,31 +1,19 @@
...
-		e *= -1
-		println("executed")
-		return 1 / fastExponentiation(b, e)
-	}
-	r := 1
-	for e > 0 {
-		if e%2 == 1 {
-			r = r * b
+func fastExponentiation(x int, y int) float64 {
+	res := 1 // res is the final result
```
Comment (nomisrevol) [inline math/fastExponent.go:4]: Remove unnecessary comments. Please add more references to the implementation (describe how it work, or refer to other document explain how it work. Naming things meaningful also. 

Could you remove the main function and write some testcases for this implementation?

**Review by nomisrevol — CHANGES_REQUESTED** [review]

**Review by vedantmamgain — COMMENTED** [review]
Check it now..

**Comment by nomisrevol** [comment]
You're giving us 3 implementations of the same algorithm. It is really good, but it will be better if in the pull request, you can show us the benchmark between these 3 implementations. Or maybe describe pros and cons of each implementation in PR. Thank you.

**Review by vedantmamgain — COMMENTED** [review]
Have added the benchmarking results of all the three types of implementations of the algorithm.

**Comment by nomisrevol** [comment]
LGTM!

**Review by nomisrevol — APPROVED** [review]

**Comment by vedantmamgain** [comment]
Merge? Or is there anything left?

**Comment by nomisrevol** [comment]
What's `primalityTest.go`? If it is an implementation for another algorithm, please cherry-pick that commit to other branch and create another pull request. Reset the HEAD to commit `e0216bb` then I will merge it. Sorry for the late response.

**Comment by vedantmamgain** [comment]
Let me make a fresh pr for both of them seperately.

**Pull request closed without merging**